docs/ai/how-to/app-service-db-auth.md

```diff
@@ -0,0 +1,314 @@
+1. On the **Review + assign** tab, select **Review + assign** to assign the role.
+
+:::zone-end
+
+:::zone target="docs" pivot="azure-cli"
+
+**Resource scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "{myVectorDbReaderRole}" \
+--scope "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceSubType}/{resourceName}"
+```
+
+**Resource group scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "{myVectorDbReaderRole}" \
+--scope "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}"
+```
+
+**Subscription scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "{myVectorDbReaderRole}" \
+--scope "/subscriptions/{subscriptionId}"
+```
+
+**Management group scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "{myVectorDbReaderRole}" \
+--scope "/providers/Microsoft.Management/managementGroups/{managementGroupName}"
+```
+
+:::zone-end
+
+### Implement token-based authentication with the vector database
+
+The following code samples require these additional libraries:
+
+* [`Azure.Identity` NuGet package](https://www.nuget.org/packages/Azure.Identity)
+* [`Microsoft.SemanticKernel.Connectors.AzureAISearch` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel.Connectors.AzureAISearch)
+
+1. Initialize a `DefaultAzureCredential` object to pick up your app's managed identity:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="tokenCredential":::
+
+1. Initialize an `IMemoryStore` object for your vector database, then use it to build an `ISemanticTextMemory`:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="aiStore":::
+
+1. Build a `Kernel` object, then import the `ISemanticTextMemory` object using the `TextMemoryPlugin`:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="addMemory":::
+
+1. Use the `Kernel` object to invoke a prompt that includes memory recall:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="useMemory":::
+
+## Use Key Vault to store connection secrets
+
+If a vector database doesn't support Microsoft Entra authentication, you can use a Key Vault to store your connection secrets and retrieve them with your App Service application. By using a Key Vault to store your connection secrets you can share them with multiple applications, and control access to individual secrets per application.
+
+Before following these steps, retrieve a connection string for your vector database. For example, see [Use Azure Cache for Redis in .NET Framework](/azure/azure-cache-for-redis/cache-dotnet-how-to-use-azure-redis-cache#retrieve-host-name-ports-and-access-keys-from-the-azure-portal).
```

Comment:
Can we link to the asp.net core version of a non .NET framework option since that is more legacy?

Maybe here: https://review.learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-web-app-aspnet-core-howto?branch=main

Reply:
yep, updated link